SpringBootCrudApi/src/main/java/com/cengenes/configuration/api/controller/BookController.java

```diff
@@ -0,0 +1,103 @@
+package com.cengenes.configuration.api.controller;
+
+import com.cengenes.configuration.api.entity.book.Book;
+import com.cengenes.configuration.api.entity.book.BookResponse;
+import com.cengenes.configuration.api.entity.response.OperationResponse;
+import com.cengenes.configuration.api.repository.BookRepository;
+import com.cengenes.configuration.api.service.AuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+
+@RestController
+@RequestMapping(value = "/api", produces = MediaType.APPLICATION_JSON_VALUE)
+@CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "*")
+public class BookController {
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Autowired
+    private AuthService authService;
+
+    @RequestMapping(value = "/books", method = RequestMethod.GET)
+    public BookResponse getProductsByPage(
+            @RequestParam(value = "page", defaultValue = "0", required = false) Integer page,
+            @RequestParam(value = "size", defaultValue = "20", required = false) Integer size,
+            @RequestParam(value = "bookId", required = false) Integer bookId,
+            @RequestParam(value = "category", required = false) String category,
+            @RequestHeader(value = "Authorization") String authorization, HttpServletResponse response,
+            Pageable pageable
+    ) {
+        authService.authenticate(authorization, response);
+        BookResponse resp = new BookResponse();
+        if (response.getStatus() == 200) {

```

Comment:
In my opinion,controller classes must work like routers.All business logic can live inside service class and all layer can talk each other from top to buttom like controller->service->repository.So,you can move all business inside service class.